cpp/src/io/json/json_normalization.cu

```diff
@@ -326,32 +328,114 @@ void normalize_single_quotes(datasource::owning_buffer<rmm::device_buffer>& inda
   std::swap(indata, outdata);
 }
 
-void normalize_whitespace(datasource::owning_buffer<rmm::device_buffer>& indata,
-                          rmm::cuda_stream_view stream,
-                          rmm::device_async_resource_ref mr)
+std::
+  tuple<rmm::device_uvector<char>, rmm::device_uvector<size_type>, rmm::device_uvector<size_type>>
+  normalize_whitespace(device_span<char const> d_input,
+                       rmm::device_uvector<size_type>& col_lengths,
+                       rmm::device_uvector<size_type>& col_offsets,
+                       rmm::cuda_stream_view stream,
+                       rmm::device_async_resource_ref mr)
 {
-  static constexpr std::int32_t min_out = 0;
-  static constexpr std::int32_t max_out = 2;
-  auto parser =
-    fst::detail::make_fst(fst::detail::make_symbol_group_lut(normalize_whitespace::wna_sgs),
-                          fst::detail::make_transition_table(normalize_whitespace::wna_state_tt),
-                          fst::detail::make_translation_functor<SymbolT, min_out, max_out>(
-                            normalize_whitespace::TransduceToNormalizedWS{}),
-                          stream);
+  size_t col_lengths_size = col_lengths.size();
+  size_type inbuf_size =
+    thrust::reduce(rmm::exec_policy_nosync(stream), col_lengths.begin(), col_lengths.end());
```

Comment:
Is this the same as `d_input.size()`?

Reply:
No, `d_input` is the raw JSON input string from which we want to only select rows of the nested column using the passed `col_lengths` and `col_offsets`. Size of `inbuf` will be much smaller than `d_input`.